tools/mo/openvino/tools/mo/back/ResultRename.py

```diff
@@ -46,4 +46,7 @@ def find_and_replace_pattern(self, graph: Graph):
                     log.warning("Tensor name for Result node with name {} wasn't found. "
                                 "Default renaming was used: {}".format(node.soft_get('name', node.id),
                                                                        result_name))
+                for idx in range(len(graph.outputs_order)):
+                    if graph.outputs_order[idx] == node['name']:
+                        graph.outputs_order[idx] = result_name
```

Comment:
I think it is better to create a dictionary prev_result_name -> new_result_name. After that you can walk though all graph.outputs_order only ones and rename them using the dictionary.